src/main/kotlin/app/batch/Validator.kt

```diff
@@ -33,60 +38,47 @@ class Validator {
                 return DecryptedRecord(jsonObject, manifestRecord)
             }
         } catch (e: Exception) {
-            val ex = BadDecryptedDataException(hbaseRowId, db, collection, e.message!!)
-            logger.error(ex.message)
-            throw ex
+            logError(logger, "Error decrypting record", e, "hbase_row_id", hbaseRowId, "db_name", db, "collection_name", collection)
+            throw BadDecryptedDataException(hbaseRowId, db, collection, e.message!!)
         }
         return null
     }
 
     fun parseDecrypted(decrypted: String): JsonObject? {
         try {
-            val jsonObject = Gson().fromJson(decrypted, JsonObject::class.java)
-            return jsonObject
+            return Gson().fromJson(decrypted, JsonObject::class.java)
         } catch (e: Exception) {
-            val parsingException = "Exception occurred while parsing decrypted db object"
             throw Exception(parsingException)
         }
     }
 
     fun retrieveId(jsonObject: JsonObject): JsonObject? {
-        val id = jsonObject.getAsJsonObject("_id")
-        if (null == id) {
-            val idNotFound = "id not found in the decrypted db object"
-            throw Exception(idNotFound)
-        }
-        return id
+        return jsonObject.getAsJsonObject("_id") ?: throw Exception(idNotFound)
     }
 
 
     fun timestampAsLong(lastUpdatedTimestamp: String): Long {
-
-        val validTimestamps = listOf("yyyy-MM-dd'T'HH:mm:ss.SSSZZZZ", "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
-
         validTimestamps.forEach {
             try {
                 val df = SimpleDateFormat(it)
                 return df.parse(lastUpdatedTimestamp).time
-            }
-            catch (e: Exception) {
-                logger.debug("'$lastUpdatedTimestamp' did not match date format '$it'")
+            } catch (e: Exception) {
+                logDebug(logger, "lastUpdatedTimestamp did not match valid formats", "valid_formats", "$validTimestamps")
```

Comment:
This log should have a tuple for "lastUpdatedTimestamp" and a tuple for the specific format being checked in this go round the loop ("$it") rather than the whole valid_formats list.

Reply:
done